pkg/manager/member/tidbcluster_status_manager.go

```diff
@@ -125,6 +136,92 @@ func (m *TidbClusterStatusManager) syncDashboardMetricStorage(tc *v1alpha1.TidbC
 	return nil
 }
 
+// ref https://github.com/pingcap/tidb/blob/36b04d1aa01db722b3f07af759168c6b8da33801/domain/infosync/info.go#L72
+// search `TopologyInformationPath` about how the key with 'ttl' and 'info' suffix is updated in that file.
+func getStaleTidbInfoKey(ctx context.Context, client pdapi.PDEtcdClient) (staleKeys []*pdapi.KeyValue, err error) {
+	kvs, err := client.Get(tidbPrefix, true /*prefix*/)
+	if err != nil {
+		return nil, perrors.AddStack(err)
+	}
+
+	infos := make(map[string]*pdapi.KeyValue)
+	ttls := make(map[string]*pdapi.KeyValue)
+
+	for _, kv := range kvs {
+		key := kv.Key
+		klog.V(4).Infof("Get TiDB info key %s", key)
+
+		if strings.HasSuffix(key, "ttl") {
+			ttls[key] = kv
+		} else if strings.HasSuffix(key, "info") {
+			infos[key] = kv
+		}
+	}
+
+	for key, kv := range infos {
+		if _, ok := ttls[strings.ReplaceAll(key, "/info", "/ttl")]; ok {
+			continue
+		}
+
+		staleKeys = append(staleKeys, kv)
+	}
+
+	return
+}
+
+// /topology/tidb/basic-tidb-0.basic-tidb-peer.tidb-cluster.svc:4000/info -> basic-tidb-0.basic-tidb-peer.tidb-cluster.svc
+func getTidbAddr(key string) (addr string) {
+	key = strings.TrimPrefix(key, tidbPrefix)
+	key = strings.TrimPrefix(key, "/")
+	tmp := strings.Split(key, ":")
+
+	addr = tmp[0]
+	return
+}
+
+func (m *TidbClusterStatusManager) syncTiDBInfoKey(tc *v1alpha1.TidbCluster) error {
+	pdEtcdClient, err := m.deps.PDControl.GetPDEtcdClient(pdapi.Namespace(tc.Namespace), tc.Name, tc.IsTLSClusterEnabled())
+	if err != nil {
+		return err
+	}
+
+	defer pdEtcdClient.Close()
+
+	kvs, err := getStaleTidbInfoKey(context.TODO(), pdEtcdClient)
```

Comment:
The kvs here should only include the keys for this TiDB cluster, otherwise, the following keys may be deleted incorrectly in L214:

- The TiDB address of the TiDB cluster with the same name in another namespace
- The TiDB address of the TiDB cluster with the same name and namespace but with a different cluster domain when deployed across Kubernetes.